### PR TITLE
cpu/cc2538: clock cleanups

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR cleanups clock initialization code for `cc2538`. It also documents the configuration in `cc2538` based boards `periph_conf`. Those boards share the same configuartion, I didn't merge it since one could use different configurations depending on the `BOARD`.

I also set `OSC_PD` before entering sleep modes if running from 32Mhz for faster wakeups.

These boards now run at `32Mhz` and use the external 32Khz oscilator.

`tests/periph_timer` is fixed to use `CLOCK_CORECLOCK` for these boards, which now accurately represents those boards clock configuration. 

### Testing procedure

**run `tests/periph_timer`:**

- master

<details><summary>BOARD=openmote-b make -C tests/periph_timer flash -j3</summary>

```
2020-03-20 16:00:09,907 # main(): This is RIOT! (Version: 2020.04-devel-1390-g7d0c4-HEAD)
2020-03-20 16:00:09,907 # 
2020-03-20 16:00:09,907 # Test for peripheral TIMERs
2020-03-20 16:00:09,910 # 
2020-03-20 16:00:09,910 # Available timers: 4
2020-03-20 16:00:09,910 # 
2020-03-20 16:00:09,911 # Testing TIMER_0:
2020-03-20 16:00:09,911 # TIMER_0: initialization successful
2020-03-20 16:00:09,913 # TIMER_0: stopped
2020-03-20 16:00:09,924 # TIMER_0: set channel 0 to 5000
2020-03-20 16:00:09,925 # TIMER_0: set channel 1 to 10000
2020-03-20 16:00:09,926 # TIMER_0: starting
2020-03-20 16:00:09,928 # TIMER_0: channel 0 fired at SW count      333 - init:      333
2020-03-20 16:00:09,939 # TIMER_0: channel 1 fired at SW count      662 - diff:      329
2020-03-20 16:00:09,939 # 
2020-03-20 16:00:09,940 # Testing TIMER_1:
2020-03-20 16:00:09,942 # TIMER_1: initialization successful
2020-03-20 16:00:09,944 # TIMER_1: stopped
2020-03-20 16:00:09,947 # TIMER_1: set channel 0 to 5000
2020-03-20 16:00:09,950 # TIMER_1: starting
2020-03-20 16:00:09,954 # TIMER_1: channel 0 fired at SW count      334 - init:      334
2020-03-20 16:00:09,954 # 
2020-03-20 16:00:09,955 # Testing TIMER_2:
2020-03-20 16:00:09,955 # TIMER_2: initialization successful
2020-03-20 16:00:09,959 # TIMER_2: stopped
2020-03-20 16:00:09,965 # TIMER_2: set channel 0 to 5000
2020-03-20 16:00:09,966 # TIMER_2: set channel 1 to 10000
2020-03-20 16:00:09,966 # TIMER_2: starting
2020-03-20 16:00:09,975 # TIMER_2: channel 0 fired at SW count      333 - init:      333
2020-03-20 16:00:09,977 # TIMER_2: channel 1 fired at SW count      662 - diff:      329
2020-03-20 16:00:09,978 # 
2020-03-20 16:00:09,986 # Testing TIMER_3:
2020-03-20 16:00:09,988 # TIMER_3: initialization successful
2020-03-20 16:00:09,991 # TIMER_3: stopped
2020-03-20 16:00:09,992 # TIMER_3: set channel 0 to 5000
2020-03-20 16:00:09,993 # TIMER_3: starting
2020-03-20 16:00:09,995 # TIMER_3: channel 0 fired at SW count      334 - init:      334
2020-03-20 16:00:10,002 # 
2020-03-20 16:00:10,002 # TEST SUCCEEDED
```
</details>

- pr

<details><summary>BOARD=openmote-b make -C tests/periph_timer flash -j3</summary>

```
2020-03-20 17:53:27,956 # Test for peripheral TIMERs
2020-03-20 17:53:27,956 # 
2020-03-20 17:53:27,956 # Available timers: 4
2020-03-20 17:53:27,956 # 
2020-03-20 17:53:27,956 # Testing TIMER_0:
2020-03-20 17:53:27,956 # TIMER_0: initialization successful
2020-03-20 17:53:27,957 # TIMER_0: stopped
2020-03-20 17:53:27,957 # TIMER_0: set channel 0 to 5000
2020-03-20 17:53:27,972 # TIMER_0: set channel 1 to 10000
2020-03-20 17:53:27,973 # TIMER_0: starting
2020-03-20 17:53:27,980 # TIMER_0: channel 0 fired at SW count      312 - init:      312
2020-03-20 17:53:27,981 # TIMER_0: channel 1 fired at SW count      620 - diff:      308
2020-03-20 17:53:27,981 # 
2020-03-20 17:53:27,982 # Testing TIMER_1:
2020-03-20 17:53:27,988 # TIMER_1: initialization successful
2020-03-20 17:53:27,988 # TIMER_1: stopped
2020-03-20 17:53:27,989 # TIMER_1: set channel 0 to 5000
2020-03-20 17:53:27,990 # TIMER_1: starting
2020-03-20 17:53:27,991 # TIMER_1: channel 0 fired at SW count      313 - init:      313
2020-03-20 17:53:27,992 # 
2020-03-20 17:53:27,992 # Testing TIMER_2:
2020-03-20 17:53:28,007 # TIMER_2: initialization successful
2020-03-20 17:53:28,010 # TIMER_2: stopped
2020-03-20 17:53:28,012 # TIMER_2: set channel 0 to 5000
2020-03-20 17:53:28,013 # TIMER_2: set channel 1 to 10000
2020-03-20 17:53:28,014 # TIMER_2: starting
2020-03-20 17:53:28,019 # TIMER_2: channel 0 fired at SW count      312 - init:      312
2020-03-20 17:53:28,021 # TIMER_2: channel 1 fired at SW count      620 - diff:      308
2020-03-20 17:53:28,021 # 
2020-03-20 17:53:28,023 # Testing TIMER_3:
2020-03-20 17:53:28,024 # TIMER_3: initialization successful
2020-03-20 17:53:28,025 # TIMER_3: stopped
2020-03-20 17:53:28,027 # TIMER_3: set channel 0 to 5000
2020-03-20 17:53:28,036 # TIMER_3: starting
2020-03-20 17:53:28,038 # TIMER_3: channel 0 fired at SW count      313 - init:      313
2020-03-20 17:53:28,038 # 
2020-03-20 17:53:28,039 # TEST SUCCEEDED
```
</details>

**run `tests/xtimer_usleep`**

- master

<details><summary>BOARD=openmote-b make -C tests/xtimer_usleep flash -j3<b> offset: 93us</b></summary>

```
2020-03-20 15:58:50,983 # main(): This is RIOT! (Version: 2020.04-devel-1390-g7d0c4-HEAD)
2020-03-20 15:58:50,999 # Running test 5 times with 7 distinct sleep times
2020-03-20 15:58:51,002 # Slept for 10093 us (expected: 10000 us) Offset: 93 us
2020-03-20 15:58:51,064 # Slept for 50093 us (expected: 50000 us) Offset: 93 us
2020-03-20 15:58:51,079 # Slept for 10327 us (expected: 10234 us) Offset: 93 us
2020-03-20 15:58:51,127 # Slept for 56873 us (expected: 56780 us) Offset: 93 us
2020-03-20 15:58:51,143 # Slept for 12215 us (expected: 12122 us) Offset: 93 us
2020-03-20 15:58:51,256 # Slept for 98858 us (expected: 98765 us) Offset: 93 us
2020-03-20 15:58:51,334 # Slept for 75093 us (expected: 75000 us) Offset: 93 us
2020-03-20 15:58:51,350 # Slept for 10093 us (expected: 10000 us) Offset: 93 us
2020-03-20 15:58:51,399 # Slept for 50093 us (expected: 50000 us) Offset: 93 us
2020-03-20 15:58:51,415 # Slept for 10327 us (expected: 10234 us) Offset: 93 us
2020-03-20 15:58:51,478 # Slept for 56873 us (expected: 56780 us) Offset: 93 us
2020-03-20 15:58:51,494 # Slept for 12215 us (expected: 12122 us) Offset: 93 us
2020-03-20 15:58:51,590 # Slept for 98858 us (expected: 98765 us) Offset: 93 us
2020-03-20 15:58:51,671 # Slept for 75093 us (expected: 75000 us) Offset: 93 us
2020-03-20 15:58:51,686 # Slept for 10093 us (expected: 10000 us) Offset: 93 us
2020-03-20 15:58:51,734 # Slept for 50093 us (expected: 50000 us) Offset: 93 us
2020-03-20 15:58:51,751 # Slept for 10327 us (expected: 10234 us) Offset: 93 us
2020-03-20 15:58:51,814 # Slept for 56873 us (expected: 56780 us) Offset: 93 us
2020-03-20 15:58:51,831 # Slept for 12215 us (expected: 12122 us) Offset: 93 us
2020-03-20 15:58:51,926 # Slept for 98858 us (expected: 98765 us) Offset: 93 us
2020-03-20 15:58:52,007 # Slept for 75093 us (expected: 75000 us) Offset: 93 us
2020-03-20 15:58:52,023 # Slept for 10093 us (expected: 10000 us) Offset: 93 us
2020-03-20 15:58:52,071 # Slept for 50093 us (expected: 50000 us) Offset: 93 us
2020-03-20 15:58:52,088 # Slept for 10327 us (expected: 10234 us) Offset: 93 us
2020-03-20 15:58:52,151 # Slept for 56873 us (expected: 56780 us) Offset: 93 us
2020-03-20 15:58:52,167 # Slept for 12215 us (expected: 12122 us) Offset: 93 us
2020-03-20 15:58:52,263 # Slept for 98858 us (expected: 98765 us) Offset: 93 us
2020-03-20 15:58:52,342 # Slept for 75093 us (expected: 75000 us) Offset: 93 us
2020-03-20 15:58:52,358 # Slept for 10093 us (expected: 10000 us) Offset: 93 us
2020-03-20 15:58:52,407 # Slept for 50093 us (expected: 50000 us) Offset: 93 us
2020-03-20 15:58:52,423 # Slept for 10327 us (expected: 10234 us) Offset: 93 us
2020-03-20 15:58:52,487 # Slept for 56873 us (expected: 56780 us) Offset: 93 us
2020-03-20 15:58:52,505 # Slept for 12215 us (expected: 12122 us) Offset: 93 us
2020-03-20 15:58:52,599 # Slept for 98858 us (expected: 98765 us) Offset: 93 us
2020-03-20 15:58:52,680 # Slept for 75093 us (expected: 75000 us) Offset: 93 us
2020-03-20 15:58:52,681 # Test ran for 1692743 us
```
</details>

- pr

<details><summary>BOARD=openmote-b make -C tests/xtimer_usleep flash -j3<b> offset: 57us</b></summary>

```
2020-03-20 15:56:19,840 # Running test 5 times with 7 distinct sleep times
2020-03-20 15:56:19,856 # Slept for 10057 us (expected: 10000 us) Offset: 57 us
2020-03-20 15:56:19,905 # Slept for 50057 us (expected: 50000 us) Offset: 57 us
2020-03-20 15:56:19,922 # Slept for 10291 us (expected: 10234 us) Offset: 57 us
2020-03-20 15:56:19,983 # Slept for 56836 us (expected: 56780 us) Offset: 56 us
2020-03-20 15:56:20,000 # Slept for 12178 us (expected: 12122 us) Offset: 56 us
2020-03-20 15:56:20,096 # Slept for 98822 us (expected: 98765 us) Offset: 57 us
2020-03-20 15:56:20,175 # Slept for 75057 us (expected: 75000 us) Offset: 57 us
2020-03-20 15:56:20,191 # Slept for 10057 us (expected: 10000 us) Offset: 57 us
2020-03-20 15:56:20,240 # Slept for 50056 us (expected: 50000 us) Offset: 56 us
2020-03-20 15:56:20,256 # Slept for 10290 us (expected: 10234 us) Offset: 56 us
2020-03-20 15:56:20,320 # Slept for 56837 us (expected: 56780 us) Offset: 57 us
2020-03-20 15:56:20,336 # Slept for 12178 us (expected: 12122 us) Offset: 56 us
2020-03-20 15:56:20,432 # Slept for 98822 us (expected: 98765 us) Offset: 57 us
2020-03-20 15:56:20,512 # Slept for 75057 us (expected: 75000 us) Offset: 57 us
2020-03-20 15:56:20,528 # Slept for 10057 us (expected: 10000 us) Offset: 57 us
2020-03-20 15:56:20,576 # Slept for 50057 us (expected: 50000 us) Offset: 57 us
2020-03-20 15:56:20,593 # Slept for 10291 us (expected: 10234 us) Offset: 57 us
2020-03-20 15:56:20,656 # Slept for 56836 us (expected: 56780 us) Offset: 56 us
2020-03-20 15:56:20,674 # Slept for 12178 us (expected: 12122 us) Offset: 56 us
2020-03-20 15:56:20,768 # Slept for 98822 us (expected: 98765 us) Offset: 57 us
2020-03-20 15:56:20,848 # Slept for 75057 us (expected: 75000 us) Offset: 57 us
2020-03-20 15:56:20,864 # Slept for 10057 us (expected: 10000 us) Offset: 57 us
2020-03-20 15:56:20,911 # Slept for 50056 us (expected: 50000 us) Offset: 56 us
2020-03-20 15:56:20,927 # Slept for 10290 us (expected: 10234 us) Offset: 56 us
2020-03-20 15:56:20,991 # Slept for 56837 us (expected: 56780 us) Offset: 57 us
2020-03-20 15:56:21,008 # Slept for 12178 us (expected: 12122 us) Offset: 56 us
2020-03-20 15:56:21,103 # Slept for 98822 us (expected: 98765 us) Offset: 57 us
2020-03-20 15:56:21,183 # Slept for 75057 us (expected: 75000 us) Offset: 57 us
2020-03-20 15:56:21,200 # Slept for 10057 us (expected: 10000 us) Offset: 57 us
2020-03-20 15:56:21,247 # Slept for 50057 us (expected: 50000 us) Offset: 57 us
2020-03-20 15:56:21,264 # Slept for 10291 us (expected: 10234 us) Offset: 57 us
2020-03-20 15:56:21,327 # Slept for 56836 us (expected: 56780 us) Offset: 56 us
2020-03-20 15:56:21,343 # Slept for 12178 us (expected: 12122 us) Offset: 56 us
2020-03-20 15:56:21,439 # Slept for 98822 us (expected: 98765 us) Offset: 57 us
2020-03-20 15:56:21,519 # Slept for 75057 us (expected: 75000 us) Offset: 57 us
2020-03-20 15:56:21,520 # Test ran for 1686186 us
```
</details>

**networking still works** (rebased on #13653)

<details><summary>BOARD=openmote-b make -C examples/xtimer_usleep</summary>

```
2020-03-20 16:03:00,506 # main(): This is RIOT! (Version: 2020.04-devel-1397-g045dc-pr_cc2538_cleanups)
2020-03-20 16:03:00,507 # RIOT network stack example application
2020-03-20 16:03:00,508 # All up, running the shell now
> ping6 fe80::3c8b:2187:3d8b:2186 -c 1000
2020-03-20 16:03:03,370 #  ping6 fe80::3c8b:2187:3d8b:2186 -c 1000
2020-03-20 16:03:03,386 # 12 bytes from fe80::3c8b:2187:3d8b:2186: icmp_seq=0 ttl=64 rssi=-41 dBm time=9.754 ms
2020-03-20 16:03:04,378 # 12 bytes from fe80::3c8b:2187:3d8b:2186: icmp_seq=1 ttl=64 rssi=-41 dBm time=9.112 ms
2020-03-20 16:03:05,386 # 12 bytes from fe80::3c8b:2187:3d8b:2186: icmp_seq=2 ttl=64 rssi=-41 dBm time=9.434 ms
```
</details>

### CODE REVIEW

**Comment by fjmolinas** [comment]
Force pushed a change making clock initialization idempotent, In case `riotboot` is added at some point.

File: boards/common/remote/include/cfg_clk_default.h [hunk 1]
```diff
@@ -27,10 +27,40 @@ extern "C" {
...
+#if SYS_CTRL_OSC_USE_XTAL
+#define CLOCK_OSC           (XOSC32M_FREQ)
+#else
+#define CLOCK_OSC           (RCOSC16M_FREQ)
+#endif
+
+#if SYS_CTRL_OSC32K_USE_XTAL
+#define CLOCK_OSC32K        (32768U)    /* XCOSC frequency */
+#else
+#define CLOCK_OSC32K        (32753U)    /* RCOSC frequency */
```
Comment (benpicco) [inline boards/common/remote/include/cfg_clk_default.h]: Should be a define (`RCOSC32K_FREQ`) just like `RCOSC16M_FREQ`

File: cpu/cc2538/cpu.c [hunk 2]
```diff
@@ -80,15 +60,41 @@ static void cpu_clock_init(void)
...
-    /* Wait for the new clock settings to take effect: */
-    while ((SYS_CTRL->cc2538_sys_ctrl_clk_sta.CLOCK_STA ^ CLOCK_CTRL_VALUE) & CLOCK_STA_MASK);
+    /* power down the oscillator not selected by OSC bit */
+    SYS_CTRL->cc2538_sys_ctrl_clk_ctrl.CLOCK_CTRLbits.OSC_PD = 1;
+    /* set desired system clock rate */
+    SYS_CTRL->cc2538_sys_ctrl_clk_ctrl.CLOCK_CTRLbits.SYS_DIV = \
+        __builtin_ctz(XOSC32M_FREQ / CLOCK_CORECLOCK);
+    /* set desired I/O clock rate */
+    SYS_CTRL->cc2538_sys_ctrl_clk_ctrl.CLOCK_CTRLbits.IO_DIV = \
+        __builtin_ctz(XOSC32M_FREQ / CLOCK_IO);
```
Comment (benpicco) [inline cpu/cc2538/cpu.c]: nice!

File: boards/common/remote/include/cfg_clk_default.h (hunk 1 above)
Comment (fjmolinas) [reply boards/common/remote/include/cfg_clk_default.h]: addressed!

**Pull request merged**